cli/start.go

```diff
@@ -440,10 +445,17 @@ func runStart(_ *cobra.Command, args []string) error {
 	}()
 
 	select {
-	case <-signalCh:
-		log.Errorf(context.TODO(), "second signal received, initiating hard shutdown")
+	case sig := <-signalCh:
+		err := fmt.Errorf("received signal '%s' during shutdown, initiating hard shutdown", sig)
+		log.Errorf(context.TODO(), "%v", err)
+		// On Unix, os.Signal is syscall.Signal and it's convertible to int.
+		ErrorCode = 128 + int(sig.(syscall.Signal))
+		return err
 	case <-time.After(time.Minute):
-		log.Errorf(context.TODO(), "time limit reached, initiating hard shutdown")
+		err := errors.New("time limit reached, initiating hard shutdown")
+		log.Errorf(context.TODO(), "%v", err)
+		ErrorCode = 2
```

Comment:
Ok here I think you need to tell me more. on Unix if you have a system service not anyone can send a signal to it. We are not sensitive to SIGHUP so I don't think the server dies just because you launch it from the command line and you close the terminal where you launched it. So in this conversation about signals we should look at how signals can be received and what it means from the sysadmin perspective:

- SIGINT: the server was launched interactively and the admin/dev presses Ctrl+C. Here I could be convinced that we want a non-zero exit code even if the shutdown was successful.

- SIGTERM, SIGQUIT: these should be only sent/received when there is a legitimate request to shut down the node. I can foresee this to be sent by system service managers / init scripts. For these you do not want a non-zero exit status to be the last thing that the service manager sees, lest you will get an error in the logs and a request for a sysadmin to investigate "why did that service didn't shut down properly".

For the timeout during shutdown definitely it should be also non-zero regardless of the other options.

Meanwhile / orthogonally, in my experience exit statuses above 128 also usually convey the connotation that the server was not able to deal with the situation gracefully. We may not want to convey this impression with cockroach if the proper shutdown was successful.

This is where my proposal comes from:

- regular shutdown initiated by soft signal or "quit:" -> exit 0
- regular shutdown initiated by Ctrl+C -> I could agree to non-zero, perhaps 130 would be ok but if the shutdown was managed without error I don't think >128 is warranted
- some other unsupported signal -> 128 +signal ok, that's what we do for SIGSEGV, SIGBUS and SIGABRT already
- timeout during shutdown -> something non-zero in any case, I'd be ok with 128+original signal
- double signal -> 128+signal code of 2nd signal